index.js

```diff
@@ -39,8 +39,13 @@ function zipFiles(files, filename, source, destination, info, verbose) {
 function pack({ source, destination, info, verbose }) {
     return packlist({ path: source })
         .then(files => {
-            return getDefaultOuputFilename({ cwd: source })
+            const destinationIsDirectory = fs.existsSync(destination) && fs.lstatSync(destination).isDirectory();
+            return getDefaultOuputFilename({ cwd: source , filename: destinationIsDirectory ? undefined : path.basename(destination)})
                 .then(filename => {
+                    if(!destinationIsDirectory) {
+                        filename = path.basename(destination);
+                        destination = path.dirname(destination);
```

Comment:
Is this stripping off the filename from the destination? If so can we move this up a level to pull apart the filename and destination before we pass it into getDefaultOutputFileName?